pkg/promclient/promclient.go

```diff
@@ -122,6 +122,36 @@ func (c *Client) get2xx(ctx context.Context, u *url.URL) (_ []byte, _ int, err e
 	return body, resp.StatusCode, nil
 }
 
+//First attempt to add ability to send POST requests [issue #2577]
+// Doubt on which piece of code required to handle parameter/flag that
+// will give the option to choose between GET and POST
+func (c *Client) post2xx(ctx context.Context, u *url.URL, skipHead bool) (_ []byte, _ int, err error) {
```

Comment:
I'd love to see this method used in action. For example, I'd probably want to pass the content that I'd like to send.